cfdp-core/src/transaction/recv.rs

```diff
@@ -66,12 +74,25 @@ pub struct RecvTransaction<T: FileStore> {
     // checksum cache to reduce I/0
     // doubles as stored checksum in received mode
     checksum: Option<u32>,
-    // tracker to know what to do when a timeout occurs
-    waiting_on: WaitingOn,
     // The current state of the transaction.
     // Used to determine when the thread should be killed
     state: TransactionState,
+    // recv sub-state, applicable when state = Active
+    recv_state: RecvState,
+    // File size received in the EOF, used also as an indication that EOF has been received
+    file_size: Option<u64>,
+    // EOF ack prepared to be sent at the next opportunity
+    ack: Option<PositiveAcknowledgePDU>,
+    // Finished PDU prepared to be sent at the next opportunity, if the bool is true
+    finished: Option<(Finished, bool)>,
+    // Prompt PDU to be processed at the next opportunity
+    prompt: Option<PromptPDU>,
+    // the list of gaps to include in the next NAK
+    naks: VecDeque<SegmentRequestForm>,
```

Comment:
If we address the TODO there,  the drain will be replaced with a loop which pops from the front as long as the resulting NAK PDU size does not exceed the maximum size. But note that I did not find the max size of the PDU defined anywhere;
I guess it could be worked out from the transaction_config.file_size_segment, not sure if that was your intention.

Reply:
I think it was, since in theory you could use that to compute the maximum PDU size. But if we want to add a parameter to track max PDU size more easily that works for me too. Then we could either pop off the front in a loop like you suggest, or since the SegmentRequestForms have a fixed size based on the large_file_flag we could use that too just drain up to what would make the max PDU size.

Reply:
Actually somehow related to this length, I want to bring out another thing for your consideration: one of my envisaged use cases would be to have a kind of "tailing file transfer" - used to transfer log files (or other streaming files) which are being appended to, so transfer bits from them as they appear (basically like a "tail -f" via CFDP). The transaction can run for days and the EOF will be sent only after I determined that the file is closed (for example rotated by logrotate).

I haven't looked yet what implications this would have in the code but one of them is to define a minimum file size segment which will be used to avoid sending data when there is just a few bytes appended to the file (except after the file has been closed).

Do you think that's a crazy idea?